### PR TITLE
InstEmitSimdHelper: Add SSE2 impl of saturate float to 32-bit int

### PR DESCRIPTION
**Pull request by merryhime**
Self explanatory. Requires tests for completness.

32-bit version only currently, because writing a 64-bit implementation is [not entirely trivial](https://github.com/merryhime/dynarmic/blob/76ec1afdad3835ad6e87e6b711663a5aaf2c1939/src/dynarmic/backend/x64/emit_x64_floating_point.cpp#L1500-L1528).

### CODE REVIEW

File: ARMeilleure/Instructions/InstEmitSimdHelper.cs [hunk 1]
```diff
@@ -1613,6 +1613,71 @@ public static Operand EmitFloatAbs(ArmEmitterContext context, Operand value, boo
...
+                if (unsigned)
+                {
+                    value = context.AddIntrinsic(Intrinsic.X86Minsd, value, X86GetScalar(context, 0x41efffffffe00000u));
+                    value = context.AddIntrinsic(Intrinsic.X86Maxsd, value, X86GetScalar(context, 0x0000000000000000u));
+                    value = context.AddIntrinsicLong(Intrinsic.X86Cvttsd2si, value);
+                }
+                else
+                {
+                    value = context.AddIntrinsic(Intrinsic.X86Minsd, value, X86GetScalar(context, 0x41dfffffffc00000u));
+                    // Out-of-range value for 32-bit conversion is is 0x80000000, no need for maxsd.
```
Comment (gdkchan) [inline ARMeilleure/Instructions/InstEmitSimdHelper.cs:1634]:
```suggestion
                    // Out-of-range value for 32-bit conversion is 0x80000000, no need for maxsd.
```

**Comment by merryhime** [comment]
Can we assume SSE2 as baseline and do away with the soft fallback completely?

**Comment by gdkchan** [comment]
> Can we assume SSE2 as baseline and do away with the soft fallback completely?

This is undesirable as it would make porting to new platforms more difficult (as we are forced to implement fast paths for those instructions). Plus the fallback already exists anyway, so why remove it?

**Comment by gdkchan** [comment]
Just mirrorring what I said on Discord here, I think it is fine to introduce `ConvertFPToSI` and `ConvertFPToUI` IR instructions for this, because float -> int conversion is a "basic" operation. Such an instruction would follow the Arm behaviour, where out of range values are clamped to the maximum allowed range. This would also eliminate the need for the "software fallback" when SSE2 is not supported (non-x86 CPUs).

**Comment by merryhime** [comment]
Revisit later. Integer saturation instructions are higher priority.

**Pull request closed without merging**